constraints_test.go

```diff
@@ -72,9 +72,21 @@ func TestConstraintCheck(t *testing.T) {
 		{"<=1.1", "0.1.0", true},
 		{"<=1.1", "1.1.0", true},
 		{"<=1.1", "1.1.1", false},
-		{">0", "0.0.1-alpha", true},
-		{">=0", "0.0.1-alpha", true},
 		{">0", "0", false},
+		{">0", "0.0.1-alpha", false},
+		{">0.0", "0.0.1-alpha", false},
+		{">0-0", "0.0.1-alpha", true},
+		{">0.0-0", "0.0.1-alpha", true},
+		{">0", "0.0.0-alpha", false},
+		{">0-0", "0.0.0-alpha", true},
+		{">=0", "0.0.1-alpha", false},
+		{">=0.0", "0.0.1-alpha", false},
+		{">=0-0", "0.0.1-alpha", true},
+		{">=0.0-0", "0.0.1-alpha", true},
+		{">=0", "0.0.0-alpha", false},
+		{">=0-0", "0.0.0-alpha", true},
+		{"<0", "0.0.0-alpha", false},
```

Comment:
Confused about this test (`{"<0", "0.0.0-alpha", false}`), especially because `{">=0", "0.0.0-alpha", false}`

Reply:
Without a pre-release in the comparator (e.g., `<0`, `>=0`) you'll find all pre-release values are ignored. That's why they are false.